pkg/occlient/occlient.go

```diff
@@ -1105,6 +1113,49 @@ func (c *Client) WaitForBuildToFinish(buildName string) error {
 	return nil
 }
 
+// WaitAndGetDC block and waits until the DeploymentConfig has updated it's annotation
+func (c *Client) WaitAndGetDC(name string, field string, value string, timeout time.Duration) (*appsv1.DeploymentConfig, error) {
+	glog.V(4).Infof("Waiting for DeploymentConfig %s annotation '%s' to update to '%s'", name, field, value)
+
+	w, err := c.appsClient.DeploymentConfigs(c.Namespace).Watch(metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", name),
+	})
+	defer w.Stop()
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to watch dc")
+	}
+
+	timeoutChannel := time.After(timeout)
+	// Keep trying until we're timed out or got a result or got an error
+	for {
+		select {
+
+		// Timout after X amount of seconds
+		case <-timeoutChannel:
+			return nil, errors.New("Timed out waiting for annotation to update")
+
+		// Each loop we check the result
+		case val, ok := <-w.ResultChan():
+
+			if !ok {
+				break
```

Comment:
should we return here?

Reply:
Yeah, you're right.

Reply:
Actually, this is okay. Since we implement the timeout anyways. Something it will return an error when the DC / annotation isn't quite ready yet.

Reply:
Ok :+1: